lib/core/src/recover/handlers/handle_send_swap.rs

```diff
@@ -61,25 +61,30 @@ impl SendSwapHandler {
         let mut recovered_data = Self::recover_onchain_data(&context.tx_map, &swap_id, history)?;
 
         // Recover preimage if needed
-        if let (Some(claim_tx_id), None) = (&recovered_data.claim_tx_id, &send_swap.preimage) {
-            match Self::recover_preimage(
-                context,
-                claim_tx_id.txid,
-                &swap_id,
-                context.swapper.clone(),
-            )
-            .await
-            {
-                Ok(Some(preimage)) => {
-                    recovered_data.preimage = Some(preimage);
-                }
-                Ok(None) => {
-                    warn!("No preimage found for Send Swap {swap_id}");
-                    recovered_data.claim_tx_id = None;
-                }
-                Err(e) => {
-                    error!("Failed to recover preimage for swap {swap_id}: {e}");
-                    recovered_data.claim_tx_id = None
+        if recovered_data.lockup_tx_id.is_some() && send_swap.preimage.is_none() {
+            // We can attempt to recover the preimage cooperatively after we know the
+            // lockup tx was broadcast. If we cannot recover it cooperatively,
+            // we can try to recover it from the claim tx.
+            if recovered_data.claim_tx_id.is_some() {
```

Comment:
This will not recover the preimage from the boltz endpoint in case of under the limit. Is that what we want?

Reply:
No, fixed [4caafc3](https://github.com/breez/breez-sdk-liquid/pull/863/commits/4caafc32fd5133b8cf6b306632679a70a69d055b)